Core/src/org/sleuthkit/autopsy/coreutils/AppDBParserHelper.java

```diff
@@ -0,0 +1,485 @@
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.coreutils;
+
+import java.util.Collection;
+import java.util.Collections;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.AccountFileInstance;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Account;
+import static org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE.TSK_CALLLOG;
+import static org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE.TSK_CONTACT;
+import static org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE.TSK_MESSAGE;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Relationship;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+
+/**
+ * A helper class to support modules that parse SQLite databases from mobile 
+ * apps and create artifacts.
+ */
+public final class AppDBParserHelper  {
+    
+    private static final Logger logger = Logger.getLogger(AppDBParserHelper.class.getName());
+    
+    private final AbstractFile dbAbstractFile;
+    private final String moduleName;
+    
+    private final AccountFileInstance selfAccountInstance;
+    
+    // type of accounts to be created for this App DB
+    private final Account.Type accountsType;
+    
+   
+   
+    private AppDBParserHelper(String moduleName, AbstractFile dbFile, AccountFileInstance selfAccountInstance, Account.Type accountsType ) {
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.selfAccountInstance = selfAccountInstance;
+        this.accountsType = accountsType;
+    }
+    
+    /**
+     * Constructs a AppDB parser helper for the given DB file.
+     * 
+     * This is a constructor for Apps that do not need to create any 
+     * accounts/relationships.
+     * 
+     * @param moduleName name of module parsing the DB
+     * @param dbFile db file
+     * 
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile) {
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.selfAccountInstance = null;
+        this.accountsType = null;
+    }
+    
+    /**
+     * Constructs a AppDB parser helper for the given DB file.
+     * 
+     * This is a constructor for Apps that that do not have any app specific account information 
+     * for device owner and will use  a 'Device' account in lieu.
+     * 
+     * It creates a DeviceAccount instance to use as a self account.
+     * 
+     * @param moduleName
+     * @param dbFile
+     * @param accountsType
+     * @param datasource
+     * 
+     * @throws TskCoreException 
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile, Account.Type accountsType, DataSource datasource) throws TskCoreException {
```

Comment:
We can get the datasource from AbstractFile.

Reply:
Of course.